registry/remote/auth/credential.go

```diff
@@ -38,3 +40,14 @@ type Credential struct {
 	// Reference: https://docs.docker.com/registry/spec/auth/token/
 	AccessToken string
 }
+
+// StaticCredential is a utility method for specifying static credentials that
+// are used when creating a client.
+func StaticCredential(registry string, cred Credential) func(ctx context.Context, reg string) (Credential, error) {
+	return func(_ context.Context, reg string) (Credential, error) {
+		if reg == registry {
+			return cred, nil
+		}
+		return EmptyCredential, nil
```

Comment:
Should add a new error like `ErrInvalidCredential` in `errdef` package and return here.

Reply:
I think `ErrCredentialNotFound` sounds more accurate. And it is within the `auth` scope.

Reply:
Between those 2 I will still stick to `ErrInvalidCredential`. If I am a user, `ErrCredentialNotFound` sounds like that the provided static credential is somehow saved to somewhere and cannot be found.

Reply:
Do we have to define an error variable? Why can't we just use `fmt.Errorf("invalid credential")`?

Reply:
> Between those 2 I will still stick to `ErrInvalidCredential`. If I am a user, `ErrCredentialNotFound` sounds like that the provided static credential is somehow saved to somewhere and cannot be found.

I agree with Billy.

Reply:
We need an error variable and we will use that in unit testing.